misk/src/test/kotlin/misk/metrics/backends/stackdriver/StackDriverReporterTest.kt

```diff
@@ -0,0 +1,290 @@
+        assertGauge(byType["$METRIC_PREFIX/foo/value"]!!, 42)
+    }
+
+    @Test
+    fun doubleGauge() {
+        val f: () -> Double = { 42.4 }
+        val gauge: Gauge<Any> = metrics.gauge("foo", f)
+
+        val timeSeries = reporter.toTimeSeries("foo", gauge, START_TIME, NOW)
+        val byType = timeSeries.map { it.metric.type to it }.toMap()
+        assertGauge(byType["$METRIC_PREFIX/foo/value"]!!, 42.4)
+
+    }
+
+    @Test
+    fun floatGauge() {
+        val f: () -> Float = { 42.4f }
+        val gauge: Gauge<Any> = metrics.gauge("foo", f)
+
+        val timeSeries = reporter.toTimeSeries("foo", gauge, START_TIME, NOW)
+        val byType = timeSeries.map { it.metric.type to it }.toMap()
+        assertGauge(byType["$METRIC_PREFIX/foo/value"]!!, 42.4)
+    }
+
+    @Test
+    fun byteGauge() {
+        val f: () -> Byte = { 42 }
+        val gauge: Gauge<Any> = metrics.gauge("foo", f)
+
+        val timeSeries = reporter.toTimeSeries("foo", gauge, START_TIME, NOW)
+        val byType = timeSeries.map { it.metric.type to it }.toMap()
+        assertGauge(byType["$METRIC_PREFIX/foo/value"]!!, 42)
+    }
+
+    @Test
+    fun bigDecimalGauge() {
+        val f: () -> BigDecimal = { BigDecimal.valueOf(42.4) }
+        val gauge: Gauge<Any> = metrics.gauge("foo", f)
+
+        val timeSeries = reporter.toTimeSeries("foo", gauge, START_TIME, NOW)
+        val byType = timeSeries.map { it.metric.type to it }.toMap()
+        assertGauge(byType["$METRIC_PREFIX/foo/value"]!!, 42.4)
+    }
+
+    @Test
+    fun bigIntegerGauge() {
+        val f: () -> BigInteger = { BigInteger.valueOf(42) }
+        val gauge: Gauge<Any> = metrics.gauge("foo", f)
+
+        val timeSeries = reporter.toTimeSeries("foo", gauge, START_TIME, NOW)
+        val byType = timeSeries.map { it.metric.type to it }.toMap()
+        assertGauge(byType["$METRIC_PREFIX/foo/value"]!!, 42.toDouble())
+    }
+
+    @Test
+    fun toTimeSeries() {
+        metrics.timer("fiddy").update(10, TimeUnit.MILLISECONDS)
+        metrics.timer("fiddy").update(23, TimeUnit.MILLISECONDS)
+        metrics.counter("fenty").inc(245)
+        metrics.counter("rent").inc(744)
+        metrics.counter("kent").inc(95)
+        metrics.gauge("im-basic", { 42 })
+        metrics.histogram("histy").update(65)
+        metrics.histogram("histy").update(66)
+        metrics.histogram("histy").update(67)
+        metrics.histogram("crows").update(99)
+        metrics.histogram("crows").update(102)
+
+        val timeSeries = reporter.toTimeSeries(
+                metrics.gauges,
+                metrics.counters,
+                metrics.histograms,
+                null,
+                metrics.timers,
+                START_TIME, NOW)
+
+        val names = timeSeries.map { it.metric.type to it }.toMap()
+        assertThat(names.keys).containsExactly(
+                "custom.googleapis.com/dw/my_app/us-west2/165.33.45.25/im-basic/value",
```

Comment:
Are these coming from an actual HTTP request?

Reply:
Stackdriver uses path format to identify metrics. The custom.googleapis.com/dw prefix is required to identify these as custom metrics.